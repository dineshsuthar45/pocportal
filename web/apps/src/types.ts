import { InputProps, TypographyProps } from "@mui/material";
import { CSSProperties, ChangeEvent } from "react";

export interface IThemeSchema {
  palette: any;
  typography: any;
  components?: any;
}

export interface IGlobalTableProps {
  row: any;
  list: any;
}

export interface IGlobalButtonProps {
  label: any;
  sublabel?: string;
  disabled?: boolean;
  onClick?: () => void;
  style?: CSSProperties;
  type?: string;
}

export interface IGlobalImgProps {
  img: string;
  alt: string;
  style?: CSSProperties;
}

export interface IGlobaltextFieldProps {
  id?: string;
  value: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  errmsg?: string;
  type?: string;
  disabled?: boolean;
  style?: CSSProperties;
  placeholder?: string;
  showPassword?: boolean;
  InputProps?: InputProps;
}

export interface IGlobalTypoProps {
  id?: number;
  variant: TypographyProps["variant"];
  label: string;
  sublabel?: string;
  style?: CSSProperties;
}
