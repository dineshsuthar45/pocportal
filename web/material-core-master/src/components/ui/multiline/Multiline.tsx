import { useTheme } from "@mui/material";
import MuiTextField from "@mui/material/TextField";
import classNames from "classnames";
import { v4 as uuidv4 } from "uuid";
import { MultilineProps } from "./Multiline.types";

/**
 * Further properties can be found in MUI website at https://mui.com/material-ui/react-text-field/#multiline
 */
export const MMultiline = ({
  autoFocus = false,
  color = "primary",
  disabled = false,
  error = false,
  fullWidth = false,
  id = uuidv4(),
  margin = "none",
  required = false,
  size = "medium",
  variant = "outlined",
  ...props
}: MultilineProps) => {
  const theme = useTheme();

  return (
    <MuiTextField
      className={classNames("multiline-root", theme.palette.mode)}
      multiline
      autoFocus={autoFocus}
      color={color}
      disabled={disabled}
      error={error}
      fullWidth={fullWidth}
      id={id}
      margin={margin}
      required={required}
      size={size}
      variant={variant}
      {...props}
    />
  );
};

MMultiline.defaultProps = {
  color: "primary",
  autoFocus: false,
  disabled: false,
  error: false,
  fullWidth: false,
  id: uuidv4(),
  margin: "none",
  required: false,
  size: "medium",
  variant: "outlined",
};
