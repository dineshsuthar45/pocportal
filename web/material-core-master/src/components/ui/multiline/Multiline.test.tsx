import { render } from '@testing-library/react';
import React from 'react';
import { MMultiline } from './Multiline';
import { MultilineProps } from './Multiline.types';

describe('Multiline', () => {
	const renderComponent = () => render(<MMultiline {...props} />);
	let props: MultilineProps;

	beforeEach(() => {
		props = {
			id: 'id',
			label: 'Multiline label',
			onChange: () => {},
		};
	});

	it('should render correctly', () => {
		const { container } = renderComponent();

		expect(container).toMatchSnapshot();
	});
});
