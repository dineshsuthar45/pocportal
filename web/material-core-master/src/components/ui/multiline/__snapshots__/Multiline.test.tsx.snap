// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiline should render correctly 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root multiline-root light css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for="id"
      id="id-label"
    >
      Multiline label
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <textarea
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
        id="id"
        style="height: 0px; overflow: hidden;"
      />
      <textarea
        aria-hidden="true"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
        readonly=""
        style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
        tabindex="-1"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-yjsfm1"
        >
          <span>
            Multiline label
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
